### DISCUSSION TITLE
Can not export: Error: Cannot find module 'http'

### DISCUSSION THREAD

**Discussion by aozora** [vercel/next.js#26413] [Help] [answered]
Hello,
I've setup my project with Next and DatoCms (repo here: https://github.com/aozora/next-ts-starter ), all is fine, dev and build are working, however whenever I try to generate the static html with `npm run export` I get this error *for every page*, and it also happens running with `npm start`:

Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Cannot find module 'http'
    at o (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:212)
    at /Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:352
    at Object.1.http (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:517)
    at o (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:318)
    at /Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:352
    at Object.3../_read (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:4755)
    at o (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:318)
    at r (/Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:450)
    at /Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:470
    at /Users/marcello/projects/next-ts-starter/.next/serverless/chunks/693.js:8:37

Repo here: https://github.com/aozora/next-ts-starter
Project is setup with TypeScript, Jest, Testing Library, Storybook.

However I've tried several things but I don't understand why and what ...., 

Looking at the chunk, I see it require the http module: `var http = require("http")` however the error suggests that it doesn't find the node http module ????

please can someone help ?

[Update 22/06/2021] I found that this happens only when in the next.config.js the target is 'serverless'!!

**Comment by adebiyial** [comment, marked as answer]
Hey. Is it possible to see your code?

**Reply by aozora** [reply]
sure, the repo is here: https://github.com/aozora/next-ts-starter

**Reply by adebiyial** [reply]
@aozora Did you check this out: https://nextjs.org/docs/advanced-features/static-html-export?

**Reply by aozora** [reply]
@adebiyial yes of course, but it didn't help.
I discovered that the issue happens only when in the next.config.js the target is 'serverless'!!

**Reply by adebiyial** [reply]
Great @aozora . Problem solved?

**Reply by aozora** [reply]
in a way...yes.
